### DISCUSSION TITLE
CC1101 Frequency Range

### DISCUSSION THREAD

**Discussion by smolbun** [jgromes/RadioLib#1111] [General]
Hi I made a [frequency analyzer](https://github.com/smolbun/CC1101-Frequency-Analyzer/tree/main) based on Flipper Zero code, I had to modify the library to use >= and <= instead. I modified [this](https://github.com/jgromes/RadioLib/blob/139fc7ee358536301cf1fb79e6121270827f49c7/src/modules/CC1101/CC1101.cpp#L427C17-L427C20) to get it working.

**Comment by jgromes** [comment]
Could you elaborate a bit more what exactly was going wrong? You are right that the allowed frequency check is not consistent with other modules like SX127x, however, whether the bound is inclusive or not should not cause any problems unless you are specifically requesting the limit frequency - at which point I would question whether that makes sense for the CC1101. Does it really support e.g. 300 MHz? The datasheet specifies the range, but not whether its bounds are also valid.

https://github.com/jgromes/RadioLib/blob/139fc7ee358536301cf1fb79e6121270827f49c7/src/modules/CC1101/CC1101.cpp#L425-L431

**Reply by smolbun** [reply]
getRSSI() is able to detect transmission at those specified frequency, e.g 300MHz. Without the changes, setFrequency() would throw an error for being invalid. Perhaps there's another way to suppress the error?
```cpp
  // check allowed frequency range
  if(!(((freq >= 300.0) && (freq <= 348.0)) ||
       ((freq >= 387.0) && (freq <= 464.0)) ||
       ((freq >= 779.0) && (freq <= 928.0)))) {
    return(RADIOLIB_ERR_INVALID_FREQUENCY);
  }
  ```

**Reply by jgromes** [reply]
> getRSSI() is able to detect transmission at those specified frequency, e.g 300MHz.

Be careful with the assumption that "RSSI increases, therefore this frequency is supported", for example, you can get RSSI increase at harmonic frequencies of the transmission. Outside of manufacturer-specified range, the sensitivity will likely be different etc. etc. - the behavior is not binary (either it works or doesn't), some things may work, others may not and yet others may behavew differently than you expect them to. For example, your RSSI reading can very well be off, and you would have no way to tell.

**Reply by smolbun** [reply]
Hmm, is there no correct way to test it then? I glanced the CC1101 datasheet "Frequency bands: 300-348 MHz, 387-464
MHz and 779-928 MHz" so I assume it was able to be set from the start till the end of each bands.

**Comment by smolbun** [comment]
Didn't realized you made the [changes ](https://github.com/jgromes/RadioLib/commit/214a566d9a317c8538b2b893f78376f65d02159c) already, I'll close this discussion thanks.
